### PR TITLE
Resource 221 lazy load iam user

### PR DESCRIPTION
**Pull request by sathish-progress**
### Description
Lazy Load IAM User

### Check List
Please fill box or appropriate ([x]) or mark N/A.
- [ ] New functionality includes integration tests/controls
- [ ] New Terraform resources
- [ ] Documentation provided or updated for resources 
- [ ] All Integration Tests pass
- [X] All Unit Tests pass
- [X] `rake lint` passes
- [X] All commits have been signed-off for the Developer Certificate of Origin. See <https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco>

### CODE REVIEW

File: libraries/aws_iam_user.rb [hunk 1]
```diff
@@ -12,51 +12,49 @@ class AwsIamUser < AwsResourceBase
...
+    @user_arn = user.arn
+    @user_id  = user.user_id
+    @username = opts[:user_name]
+  end
+
+  def access_keys
+    @access_keys ||= (@aws.iam_client.list_access_keys(user_params).access_key_metadata || [])
+  end
+
+  def has_console_password
```
Comment (clintoncwolfe) [inline libraries/aws_iam_user.rb:34]: I think you need to define this with a question mark at the end of the method name to maintain the semantics. Previously there was an alias, but you have deleted the alias.

Comment (clintoncwolfe) [inline libraries/aws_iam_user.rb:34]: I think you need to define this with a question mark at the end of the method name to maintain the semantics. Previously there was an alias, but you have deleted the alias.

File: libraries/aws_iam_user.rb [hunk 2]
```diff
@@ -12,51 +12,49 @@ class AwsIamUser < AwsResourceBase
...
+
+  def attached_policy_names
+    @attached_policy_names ||= policies.map { |p| p[:policy_name] }
+  end
+
+  def attached_policy_arns
+    @attached_policy_arns ||= policies.map { |p| p[:policy_arn] }
+  end
+
+  def has_mfa_enabled
```
Comment (clintoncwolfe) [inline libraries/aws_iam_user.rb:48]: Same here - I think you need a question mark here.

Comment (clintoncwolfe) [inline libraries/aws_iam_user.rb:48]: Same here - I think you need a question mark here.

**Review by clintoncwolfe — APPROVED** [review]
Looks good, only question is why there are two empty doc files for cloudfront included in the PR?

File: libraries/aws_iam_user.rb (hunk 1 above)
Comment (sathish-progress) [reply libraries/aws_iam_user.rb:34]: alias is here https://github.com/inspec/inspec-aws/pull/887/files#diff-a09b84266041b20c6dd831d23526e66c25266f94731d9a3b2d716ff7318ec275R38

File: libraries/aws_iam_user.rb (hunk 2 above)
Comment (sathish-progress) [reply libraries/aws_iam_user.rb:48]: alias is here again https://github.com/inspec/inspec-aws/pull/887/files#diff-a09b84266041b20c6dd831d23526e66c25266f94731d9a3b2d716ff7318ec275R56

**Pull request merged**